### PR TITLE
Add support for Alias route53 records

### LINKED ISSUE #36: Support DNS traffic switching using Alias records
We need to support Alias records for DNS traffic switching, i.e. we need to adapt the following code part to not only support CNAME records:
https://github.com/zalando-zmon/zmon-aws-agent/blob/master/zmon_aws_agent/aws.py#L53

Background: Senza switches from CNAME to Alias records (https://github.com/zalando-stups/senza/issues/197).

### PR DESCRIPTION
**Pull request by mohabusama**
Fix #36

### CODE REVIEW

File: zmon_aws_agent/aws.py [hunk 1]
```diff
@@ -81,8 +81,12 @@ def populate_dns_data():
             result = route53.list_resource_record_sets(**recordfilter)
             records.extend(result['ResourceRecordSets'])
 
-        DNS_RR_CACHE_ZONE[zone['Name']] = list(
-            filter(lambda x: (x['Type'] == 'CNAME') and ('SetIdentifier' in x), records))
+        DNS_RR_CACHE_ZONE[zone['Name']] = [
+            r for r in records if (
+                ('SetIdentifier' in r and 'Weight' in r) and
```
Comment (hjacobs) [inline zmon_aws_agent/aws.py]: the brackets look confusing (wrong even?) here

Comment (mohabusama) [inline zmon_aws_agent/aws.py]: It is wrong indeed, in the type and alias target condition.

File: zmon_aws_agent/aws.py [hunk 2]
```diff
@@ -84,8 +84,8 @@ def populate_dns_data():
         DNS_RR_CACHE_ZONE[zone['Name']] = [
             r for r in records if (
                 ('SetIdentifier' in r and 'Weight' in r) and
-                r['type'] == 'CNAME' or
-                ('AliasTarget' in r and 'DNSName' in r['AliasTarget']))
+                (r['type'] == 'CNAME' or 'AliasTarget' in r and 'DNSName' in r['AliasTarget'])
```
Comment (hjacobs) [inline zmon_aws_agent/aws.py]: I would shorten to:

```
 'DNSName' in  r.get('AliasTarget', {})

**Comment by hjacobs** [comment]
:+1:

**Comment by mohabusama** [comment]
👍

File: zmon_aws_agent/aws.py [hunk 3]
```diff
@@ -81,8 +81,12 @@ def populate_dns_data():
             result = route53.list_resource_record_sets(**recordfilter)
             records.extend(result['ResourceRecordSets'])
 
-        DNS_RR_CACHE_ZONE[zone['Name']] = list(
-            filter(lambda x: (x['Type'] == 'CNAME') and ('SetIdentifier' in x), records))
+        DNS_RR_CACHE_ZONE[zone['Name']] = [
+            r for r in records if (
+                ('SetIdentifier' in r and 'Weight' in r) and
+                (r['type'] == 'CNAME' or r.get('AliasTarget', {}).get('DNSName'))
```
Comment (hjacobs) [inline zmon_aws_agent/aws.py]: "type" -> "Type"

**Pull request merged**